dsp-value-objects/src/test/scala/dsp/valueobjects/IriSpec.scala

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright © 2021 - 2022 Swiss National Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.valueobjects
+
+import dsp.valueobjects.Iri._
+import zio.prelude.Validation
+import zio.test._
+
+/**
+ * This spec is used to test the [[Iri]] value objects creation.
+ */
+object IriSpec extends ZIOSpecDefault {
+
+  // make it GroupIri instead of GroupIri + for all iris
+  val invalidIri                 = "Invalid IRI"
+  val validGroupIri              = "http://rdfh.ch/groups/0803/qBCJAdzZSCqC_2snW5Q7Nw"
+  val groupIriWithUUIDVersion3   = "http://rdfh.ch/groups/0803/rKAU0FNjPUKWqOT8MEW_UQ"
+  val validListIri               = "http://rdfh.ch/lists/0803/qBCJAdzZSCqC_2snW5Q7Nw"
+  val listIriWithUUIDVersion3    = "http://rdfh.ch/lists/0803/6_xROK_UN1S2ZVNSzLlSXQ"
+  val validProjectIri            = "http://rdfh.ch/projects/0001"
+  val projectIriWithUUIDVersion3 = "http://rdfh.ch/projects/tZjZhGSZMeCLA5VeUmwAmg"
+  val validUserIri               = "http://rdfh.ch/users/jDEEitJESRi3pDaDjjQ1WQ"
+  val userIriWithUUIDVersion3    = "http://rdfh.ch/users/cCmdcpn2MO211YYOplR1hQ"
+
+  def spec = (groupIriTest + listIriTest + projectIriTest)
+
+  private val groupIriTest = suite("IriSpec - GroupIri")(
+    test("pass an empty value and throw an error") {
+      assertTrue(GroupIri.make("") == Validation.fail(V2.BadRequestException(IriErrorMessages.GroupIriMissing)))
+      assertTrue(
+        GroupIri.make(Some("")) == Validation.fail(V2.BadRequestException(IriErrorMessages.GroupIriMissing))
+      )
+    } +
+      test("pass an invalid value and throw an error") {
+        assertTrue(
+          GroupIri.make(invalidIri) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.GroupIriInvalid)
+          )
+        )
+        assertTrue(
+          GroupIri.make(Some(invalidIri)) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.GroupIriInvalid)
+          )
+        )
+      } +
+      test("pass an invalid IRI containing unsupported UUID version and throw an error") {
+        assertTrue(
+          GroupIri.make(groupIriWithUUIDVersion3) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.UuidInvalid)
+          )
+        )
+        assertTrue(
+          GroupIri.make(Some(groupIriWithUUIDVersion3)) == Validation.fail(
+            V2.BadRequestException(IriErrorMessages.UuidInvalid)
+          )
+        )
+      } +
+      test("pass a valid value and successfully create value object") {
+        assertTrue(GroupIri.make(validGroupIri).toOption.get.value == validGroupIri)
+        assertTrue(GroupIri.make(Option(validGroupIri)).getOrElse(null).get.value == validGroupIri)
+      } +
+      test("pass None") {
```

Comment:
as above: the description could be more speaking

Reply:
True that.